pypowsybl/loadflow.py

```diff
@@ -547,9 +547,35 @@ def get_provider_parameters_names(provider: str = '') -> _List[str]:
     """
     Get list of parameters for the specified loadflow provider.
 
-    If not specified the provider will be the default one.
+    Args:
+       provider (str): the provider, if not specified the provider will be the default one.
 
     Returns:
         the list of provider's parameters
     """
     return _pypowsybl.get_loadflow_provider_parameters_names(provider)
+
+
+def get_provider_parameters(provider: str = '') -> _DataFrame:
```

Comment:
for the default value should it be better to set None ?

Reply:
Done

Reply:
in the case if someone just use get_provider_parameters() or with a non existant provider is it handle on powsybl side ? I dont think it is on pypowsybl

Reply:
> Done

I replaced the empty str as default value by none which is indeed more pythonic

Reply:
@EtienneLt It looks ok:

```python
>>> import pypowsybl as pp
>>> pp.loadflow.get_provider_parameters()
                                                                                          description     type                         default                                    possible_values
name                                                                                                                                                                                             
slackBusSelectionMode                                                        Slack bus selection mode   STRING                     MOST_MESHED      [FIRST, MOST_MESHED, NAME, LARGEST_GENERATOR]
slackBusesIds                                                                           Slack bus IDs   STRING                                                                                   
lowImpedanceBranchMode                                                      Low impedance branch mode   STRING  REPLACE_BY_ZERO_IMPEDANCE_LINE  [REPLACE_BY_ZERO_IMPEDANCE_LINE, REPLACE_BY_MI...
voltageRemoteControl                                                 Generator voltage remote control  BOOLEAN                            true                                                   
throwsExceptionInCaseOfSlackDistributionFailure     Throws an exception in case of slack distribut...  BOOLEAN                           false                                                   
loadPowerFactorConstant                                                 Load power factor is constant  BOOLEAN                           false                                                   
plausibleActivePowerLimit                                                Plausible active power limit   DOUBLE                          5000.0                                                   
slackBusPMaxMismatch                                              Slack bus max active power mismatch   DOUBLE                             1.0                                                   
voltagePerReactivePowerControl                                       Voltage per reactive power slope  BOOLEAN                           false                                                   
reactivePowerRemoteControl                                          SVC remote reactive power control  BOOLEAN                           false                                                   
maxIteration                                                                           Max iterations  INTEGER                              30                                                   
newtonRaphsonConvEpsPerEq                             Newton-Raphson convergence epsilon per equation   DOUBLE                          1.0E-4                                                   
voltageInitModeOverride                                                    Voltage init mode override   STRING                            NONE            [NONE, VOLTAGE_MAGNITUDE, FULL_VOLTAGE]
transformerVoltageControlMode                                        Transformer voltage control mode   STRING  WITH_GENERATOR_VOLTAGE_CONTROL  [WITH_GENERATOR_VOLTAGE_CONTROL, AFTER_GENERAT...
shuntVoltageControlMode                                                    Shunt voltage control mode   STRING  WITH_GENERATOR_VOLTAGE_CONTROL  [WITH_GENERATOR_VOLTAGE_CONTROL, INCREMENTAL_V...
dcPowerFactor                                                           DC approximation power factor   DOUBLE                             1.0                                                   
minPlausibleTargetVoltage                                                Min plausible target voltage   DOUBLE                             0.8                                                   
maxPlausibleTargetVoltage                                                Max plausible target voltage   DOUBLE                             1.2                                                   
minRealisticVoltage                                                             Min realistic voltage   DOUBLE                             0.5                                                   
maxRealisticVoltage                                                             Max realistic voltage   DOUBLE                             1.5                                                   
reactiveRangeCheckMode                                                      Reactive range check mode   STRING                             MAX                           [MIN_MAX, MAX, TARGET_P]
lowImpedanceThreshold                                             Low impedance threshold in per unit   DOUBLE                          1.0E-8                                                   
networkCacheEnabled                                                             Network cache enabled  BOOLEAN                           false                                                   
svcVoltageMonitoring                                                           SVC voltage monitoring  BOOLEAN                            true                                                   
stateVectorScalingMode                                                      State vector scaling mode   STRING                            NONE            [NONE, LINE_SEARCH, MAX_VOLTAGE_CHANGE]
maxSlackBusCount                                                            Maximum slack buses count  INTEGER                               1                                                   
debugDir                                                                Directory to dump debug files   STRING                                                                                 []
incrementalTransformerVoltageControlOuterLoopMa...  Incremental transformer voltage control maximu...  INTEGER                               3                                                   
secondaryVoltageControl                                          Secondary voltage control simulation  BOOLEAN                           false                                                   
ReactiveLimitsMaxPqPvSwitch                                      Reactive limits maximum Pq Pv switch  INTEGER                               3                                                   
>>> pp.loadflow.get_provider_parameters('ee')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jamgotchiangeo/PycharmProjects/pypowsybl/pypowsybl/loadflow.py", line 580, in get_provider_parameters
    series_array = _pypowsybl.create_loadflow_provider_parameters_series_array('' if provider is None else provider)
pypowsybl._pypowsybl.PyPowsyblError: No loadflow provider for name 'ee'
>>> 
```